hugolib/page.go

```diff
@@ -1348,10 +1388,11 @@ func (p *Page) parse(reader io.Reader) error {
 	}
 
 	if meta != nil {
-		if err = p.update(meta); err != nil {
+		if err = p.readMetadata(meta); err != nil {
 			return err
 		}
 	}
+	p.updateMetadata()
```

Comment:
this is added so when you render pages that are missing frontmatter they don't get more broken than they need to be. Took me ages to figure out that title/slug and lastmod changes were not handled unless I had an explicit title in frontmatter. With this change it at least still sorts and gets rendered somewhat properly.